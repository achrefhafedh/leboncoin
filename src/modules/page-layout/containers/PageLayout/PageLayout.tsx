import React from 'react';
import { Route, Redirect, Switch, BrowserRouter } from 'react-router-dom';

import { loginRoutes } from 'modules/login';
import { postsRoutes } from 'modules/posts';

const defaultRedirection = () => <Redirect to="/login" />;
const routes = [...loginRoutes, ...postsRoutes];

export const PageLayout = () => {
  return (
    <BrowserRouter>
      <Switch>
        {routes.map((route, i) => (
          <Route
            key={i}
            path={route.path}
            exact={route.exact}
            component={route.content}
          />
        ))}
        <Route render={defaultRedirection} />
      </Switch>
    </BrowserRouter>
  );
};
