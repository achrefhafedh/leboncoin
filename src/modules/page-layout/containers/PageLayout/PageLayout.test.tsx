import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { PageLayout } from './PageLayout';

configure({ adapter: new Adapter() });

describe('renders without crashing', () => {
  it('render without crashing', () => {
    const pageLayout = shallow(<PageLayout />);
    expect(pageLayout).toMatchSnapshot();
  });
});
