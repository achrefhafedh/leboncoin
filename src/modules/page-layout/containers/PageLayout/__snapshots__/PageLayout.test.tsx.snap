// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/posts"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/post"
          />
          <Route
            render={[Function]}
          />
        </Switch>
      </BrowserRouter>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/posts"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/post"
          />
          <Route
            render={[Function]}
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            Array [
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/posts"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/post"
              />,
            ],
            <Route
              render={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/posts",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/post",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/posts"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/post"
            />
            <Route
              render={[Function]}
            />
          </Switch>
        </BrowserRouter>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/posts"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/post"
            />
            <Route
              render={[Function]}
            />
          </Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/posts"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/post"
                />,
              ],
              <Route
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/posts",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/post",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
